src/site/SiteNavigation.js

```diff
@@ -0,0 +1,207 @@
+import { GetFirstElementOrDefault } from "../utilities/dom.utils";
+import { SetAttribute } from "../utilities/dom.utilities";
+import FocusTrap from "focus-trap";
+
+const attributes = {
+  ariaExpanded: "aria-expanded",
+  tabIndex: "tabindex"
+};
+
+const icons = {
+  open: "fa-bars",
+  close: "fa-times"
+};
+
+const ids = {
+  siteNav: "bc_site-nav",
+  siteNavToggleButton: "bc_site-nav__toggle-button",
+  siteNavLinks: "bc_site-nav__links",
+  page: "bc_page"
+};
+
+const cssClasses = {
+  hidden: "hidden",
+  hiddenInit: "hidden-init",
+  isActive: "is-active",
+  isDisabled: "is-disabled",
+  siteNavToggleButtonText: "bc_site-nav__toggle-button__text"
+};
+
+const invisibleTabIndex = "-1"; /** Allows us to hide an item from tabbing */
+let siteNavFocusTrap;
+
+/**
+ * * Hide/Show the site navigation based on the given shouldShow param
+ * @param {boolean} shouldShow if set to true the site navigation will be shown
+ */
+const toggleSiteNav = shouldShow => {
+  const classListAction = shouldShow ? "add" : "remove";
+  const siteNavToggleButtonElm = document.getElementById(
+    ids.siteNavToggleButton
+  );
+  const buttonIconElm = GetFirstElementOrDefault(siteNavToggleButtonElm, "i");
+  const buttonIconTextElm = GetFirstElementOrDefault(
+    siteNavToggleButtonElm,
+    `.${cssClasses.siteNavToggleButtonText}`
+  );
+  const pageElm = document.getElementById(ids.page);
+
+  // Update toggle button text
+  buttonIconTextElm.textContent = shouldShow ? "Close" : "Menu";
+
+  // Toggle the toggle button icon
+  buttonIconElm.classList.remove(shouldShow ? icons.open : icons.close);
+  buttonIconElm.classList.add(shouldShow ? icons.close : icons.open);
+
```

Comment:
Is there a reason you aren't using the ToggleClass function you created in the utilities file below? Or at the very least the RemoveClass and AddClass?

Reply:
I'll need to check, but this is a good point.

Reply:
so short answer, i did not make the utilities robust enough.

Reply:
@tmccoy529 i renamed the utilities to avoid some of the confusion. The utilities target a list of things, any html element list type thing that can use `.forEach`. Any case that I am not using this is because it is a singular element and in those cases we can just set attributes or add classes is easy enough to do.

Please let me know if this makes sense.

Reply:
Yes, that makes more sense.